src/main/java/dev/slimevr/osc/VRCOSCHandler.java

```diff
@@ -0,0 +1,252 @@
+	private OSCPortOut oscSender;
+	private OSCMessage oscMessage;
+	private final OSCConfig config;
+	private final HMDTracker hmd;
+	private final WindowsNamedPipeBridge steamvrBridge;
+	private final HumanPoseProcessor humanPoseProcessor;
+	private final List<? extends ShareableTracker> shareableTrackers;
+	private final FastList<Float> oscArgs = new FastList<>(3);
+	private final Vector3f vec = new Vector3f();
+	private final Quaternion quatBuf = new Quaternion();
+	private final float[] floatBuf = new float[3];
+	private final boolean[] trackersEnabled;
+	private long timeAtLastOSCMessageReceived;
+	private static final long HMD_TIMEOUT = 15000;
+
+	public VRCOSCHandler(
+		HMDTracker hmd,
+		HumanPoseProcessor humanPoseProcessor,
+		WindowsNamedPipeBridge steamvrBridge,
+		OSCConfig oscConfig,
+		List<? extends ShareableTracker> shareableTrackers
+	) {
+		this.hmd = hmd;
+		this.humanPoseProcessor = humanPoseProcessor;
+		this.steamvrBridge = steamvrBridge;
+		this.config = oscConfig;
+		this.shareableTrackers = shareableTrackers;
+
+		trackersEnabled = new boolean[shareableTrackers.size()];
+
+		refreshSettings();
+	}
+
+	public void refreshSettings() {
+		// Sets which trackers are enabled and force HEAD to false
+		for (int i = 0; i < shareableTrackers.size(); i++) {
+			if (shareableTrackers.get(i).getTrackerRole() != TrackerRole.HEAD) {
+				trackersEnabled[i] = config
+					.getOSCTrackerRole(shareableTrackers.get(i).getTrackerRole(), false);
+			}
+		}
+
+		// Stops listening and closes OSC port
+		if (oscReceiver != null && oscReceiver.isListening()) {
+			oscReceiver.stopListening();
+		}
+		if (oscSender != null && oscSender.isConnected()) {
+			try {
+				oscSender.close();
+			} catch (IOException e) {
+				LogManager.warning("[VRCOSCHandler] Error closing the OSC sender: " + e);
+			}
+		}
+
+		if (config.getEnabled()) {
+			// Instantiates the OSC receiver
+			try {
+				int port = config.getPortIn();
+				oscReceiver = new OSCPortIn(
+					port
+				);
+				LogManager.info("[VRCOSCHandler] Listening to port " + port);
+			} catch (IOException e) {
+				LogManager
+					.warning("[VRCOSCHandler] Error listening to the port " + config.getPortIn());
+			}
+
+			// Starts listening for the Upright parameter from VRC
+			OSCMessageListener listener = this::handleReceivedMessage;
+			MessageSelector selector = new OSCPatternAddressMessageSelector(
+				"/avatar/parameters/Upright"
+			);
+			oscReceiver.getDispatcher().addListener(selector, listener);
+			oscReceiver.startListening();
+
+			// Instantiate the OSC sender
+			try {
+				InetAddress address = config.getAddress();
+				int port = config.getPortOut();
+				oscSender = new OSCPortOut(
+					address,
+					port
+				);
+				oscSender.connect();
+				LogManager
+					.info(
+						"[VRCOSCHandler] Sending to port "
+							+ port
+							+ " at address "
+							+ address.toString()
+					);
+			} catch (IOException e) {
+				LogManager
+					.warning(
```

Comment:
This should be severe
```suggestion
					.severe(
```